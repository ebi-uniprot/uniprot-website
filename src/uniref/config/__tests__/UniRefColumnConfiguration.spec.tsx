import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import UniProtKBColumnConfiguration from '../UniRefColumnConfiguration';
import renderWithRedux from '../../../shared/__test-helpers__/RenderWithRedux';
jest.mock('../../../tools/utils/storage');

const data = {
  id: 'UniRef90_Q28I25',
  name: 'Cluster: CDC7 cell division cycle 7',
  entryType: 'UniRef90',
  sequenceLength: 483,
  memberCount: 9,
  memberIdTypes: ['UniParc', 'UniProtKB Unreviewed (TrEMBL)'],
  members: [
    'Q28I25',
    'A0A6I8RXP7',
    'A0A6I8SV39',
    'A0A6I8QDI3',
    'A0A6I8QHP6',
    'B7ZUP5',
    'UPI0012F6CEF4',
    'UPI0012F702BF',
    'UPI0012F66615',
  ],
  commonTaxon: 'Xenopus tropicalis (Western clawed frog) (Silurana tropicalis)',
  commonTaxonId: 8364,
  organismIds: [8364],
  organisms: ['Xenopus tropicalis (Western clawed frog) (Silurana tropicalis)'],
};

describe('UniRefColumnConfiguration component', () => {
  test('should render all columns', () => {
    UniProtKBColumnConfiguration.forEach((column) => {
      const { asFragment } = renderWithRedux(
        <MemoryRouter>{column.render(data)}</MemoryRouter>
      );
      expect(asFragment()).toMatchSnapshot(column.label);
    });
  });
});
