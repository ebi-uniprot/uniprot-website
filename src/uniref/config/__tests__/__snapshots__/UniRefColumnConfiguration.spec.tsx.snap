// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniRefColumnConfiguration component should render column "common_taxon": common_taxon 1`] = `
<DocumentFragment>
  <a
    href="/taxonomy/1305"
    title="Streptococcus sanguinis, taxon ID 1305"
    translate="no"
  >
    Streptococcus sanguinis
  </a>
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "common_taxonid": common_taxonid 1`] = `
<DocumentFragment>
  <a
    href="/taxonomy/1305"
    title="Streptococcus sanguinis, taxon ID 1305"
    translate="no"
  >
    1305
  </a>
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "count": count 1`] = `
<DocumentFragment>
  1 member
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "created": created 1`] = `
<DocumentFragment>
  <time
    datetime="2021-02-10T00:00:00.000Z"
  >
    2021-02-10
  </time>
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "from": from 1`] = `<DocumentFragment />`;

exports[`UniRefColumnConfiguration component should render column "id": id 1`] = `
<DocumentFragment>
  <span
    class="no-wrap accession-view"
  >
    <a
      class="accession"
      href="/uniref/UniRef100_A0A0B7GQ86"
    >
      UniRef100_A0A0B7GQ86
    </a>
  </span>
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "identity": identity 1`] = `
<DocumentFragment>
  UniRef100
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "length": length 1`] = `
<DocumentFragment>
  730
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "members": members 1`] = `
<DocumentFragment>
  <ul
    class="no-bullet"
  >
    <li>
      <a
        href="/uniprotkb/A0A0B7GQ86"
      >
        A0A0B7GQ86
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "name": name 1`] = `
<DocumentFragment>
  Cluster: PTS system glucose-specific EIICBA component [includes: glucose permease IIC component glucose-specific phosphotransferase enzyme IIB component glucose-specific phosphotransferase enzyme IIA compon...
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "organism": organism 1`] = `
<DocumentFragment>
  <ul
    class="no-bullet"
  >
    <li>
      <a
        href="/taxonomy/1305"
        title="Streptococcus sanguinis, taxon ID 1305"
        translate="no"
      >
        Streptococcus sanguinis
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "organism_id": organism_id 1`] = `
<DocumentFragment>
  <ul
    class="no-bullet"
  >
    <li>
      <a
        href="/taxonomy/1305"
        title="Streptococcus sanguinis, taxon ID 1305"
        translate="no"
      >
        1305
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "sequence": sequence 1`] = `
<DocumentFragment>
  <div>
    <section
      class="sequence-container"
    >
      <div
        class="sequence"
      >
        <span
          class="sequence__chunk"
        >
          MMKDSFKNIF
        </span>
        <span
          class="sequence__chunk"
        >
          SFEFWQKFGK
        </span>
        <span
          class="sequence__chunk"
        >
          ALMVVVAVMP
        </span>
        <span
          class="sequence__chunk"
        >
          AAGLMISIGK
        </span>
        <span
          class="sequence__chunk"
        >
          SIPMINPNLG
        </span>
        <span
          class="sequence__chunk"
        >
          VLVTTGGVLE
        </span>
        <span
          class="sequence__chunk"
        >
          QIGWGVIGNL
        </span>
        <span
          class="sequence__chunk"
        >
          HILFALAIGG
        </span>
        <span
          class="sequence__chunk"
        >
          SWAKERAGGA
        </span>
        <span
          class="sequence__chunk"
        >
          FAAGLSFILI
        </span>
        <span
          class="sequence__chunk"
        >
          NRITGVMFGV
        </span>
        <span
          class="sequence__chunk"
        >
          TGDMLADKTA
        </span>
        <span
          class="sequence__chunk"
        >
          VVKTMFGASI
        </span>
        <span
          class="sequence__chunk"
        >
          KVSDYFISVL
        </span>
        <span
          class="sequence__chunk"
        >
          ESPALNMGVF
        </span>
        <span
          class="sequence__chunk"
        >
          VGIIAGFVGA
        </span>
        <span
          class="sequence__chunk"
        >
          TAYNKYYNFR
        </span>
        <span
          class="sequence__chunk"
        >
          KLPDALSFFN
        </span>
        <span
          class="sequence__chunk"
        >
          GKRFVPFVVI
        </span>
        <span
          class="sequence__chunk"
        >
          LRSAIVAIVL
        </span>
        <span
          class="sequence__chunk"
        >
          SFVWPVVQSG
        </span>
        <span
          class="sequence__chunk"
        >
          INSFGIWIAN
        </span>
        <span
          class="sequence__chunk"
        >
          SQDTAPILAP
        </span>
        <span
          class="sequence__chunk"
        >
          FIYGTLERLL
        </span>
        <span
          class="sequence__chunk"
        >
          LPFGLHHMLT
        </span>
        <span
          class="sequence__chunk"
        >
          IPMNYTELGG
        </span>
        <span
          class="sequence__chunk"
        >
          VYEVITGSGA
        </span>
        <span
          class="sequence__chunk"
        >
          GTTVAGQDPL
        </span>
        <span
          class="sequence__chunk"
        >
          WLAWVTDLVG
        </span>
        <span
          class="sequence__chunk"
        >
          TKTADPNTYK
        </span>
        <span
          class="sequence__chunk"
        >
          HLLETVHPAR
        </span>
        <span
          class="sequence__chunk"
        >
          FKVGQMIGSF
        </span>
        <span
          class="sequence__chunk"
        >
          GILMGVAAAI
        </span>
        <span
          class="sequence__chunk"
        >
          YHNVDADKKH
        </span>
        <span
          class="sequence__chunk"
        >
          KYKGMMIATA
        </span>
        <span
          class="sequence__chunk"
        >
          LATFLTGVTE
        </span>
        <span
          class="sequence__chunk"
        >
          PIEYMFMFVA
        </span>
        <span
          class="sequence__chunk"
        >
          TPLYLVYSLV
        </span>
        <span
          class="sequence__chunk"
        >
          QGAAFAMADI
        </span>
        <span
          class="sequence__chunk"
        >
          VALRVHSFGS
        </span>
        <span
          class="sequence__chunk"
        >
          IEFLTRTPMA
        </span>
        <span
          class="sequence__chunk"
        >
          INAGLGGDII
        </span>
        <span
          class="sequence__chunk"
        >
          NFIWVTILFG
        </span>
        <span
          class="sequence__chunk"
        >
          VVMYFISNFM
        </span>
        <span
          class="sequence__chunk"
        >
          IKKFNYATPG
        </span>
        <span
          class="sequence__chunk"
        >
          RNGNYETAEG
        </span>
        <span
          class="sequence__chunk"
        >
          SDEASSSDST
        </span>
        <span
          class="sequence__chunk"
        >
          GGKVAAASQA
        </span>
        <span
          class="sequence__chunk"
        >
          VNVINLLGGR
        </span>
        <span
          class="sequence__chunk"
        >
          ANIVDVDACM
        </span>
        <span
          class="sequence__chunk"
        >
          TRLRVTVKDA
        </span>
        <span
          class="sequence__chunk"
        >
          EKVGTEEQWK
        </span>
        <span
          class="sequence__chunk"
        >
          AEGAMGLVMK
        </span>
        <span
          class="sequence__chunk"
        >
          GQGVQAIYGP
        </span>
        <span
          class="sequence__chunk"
        >
          KADVLKSDIQ
        </span>
        <span
          class="sequence__chunk"
        >
          DLLDSGEVIP
        </span>
        <span
          class="sequence__chunk"
        >
          ETLPSQKAES
        </span>
        <span
          class="sequence__chunk"
        >
          EAAEVSYKGV
        </span>
        <span
          class="sequence__chunk"
        >
          TEEVETVADG
        </span>
        <span
          class="sequence__chunk"
        >
          QVIDLADVKD
        </span>
        <span
          class="sequence__chunk"
        >
          PVFSQKMMGD
        </span>
        <span
          class="sequence__chunk"
        >
          GFAVEPENGK
        </span>
        <span
          class="sequence__chunk"
        >
          IYSPVAGTVT
        </span>
        <span
          class="sequence__chunk"
        >
          SVFPSKHAIG
        </span>
        <span
          class="sequence__chunk"
        >
          LVTDNGLEVL
        </span>
        <span
          class="sequence__chunk"
        >
          VHIGLETVSL
        </span>
        <span
          class="sequence__chunk"
        >
          EGKPFEVHVS
        </span>
        <span
          class="sequence__chunk"
        >
          EGQKVAAGDL
        </span>
        <span
          class="sequence__chunk"
        >
          LVTADLEAIK
        </span>
        <span
          class="sequence__chunk"
        >
          EAGRETSTIV
        </span>
        <span
          class="sequence__chunk"
        >
          VFTNAAAIKS
        </span>
        <span
          class="sequence__chunk"
        >
          VTVEKLGQAS
        </span>
        <span
          class="sequence__chunk sequence__chunk--display-last"
        >
          AKTVVAKVEL
        </span>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render column "types": types 1`] = `
<DocumentFragment>
  <span
    class="entry-title__status icon--unreviewed"
    title="This marks an unreviewed UniProtKB entry"
  >
    <test-file-stub />
  </span>
</DocumentFragment>
`;
